addons/google_account/google_account.py

```diff
@@ -1,29 +1,29 @@
 # -*- coding: utf-8 -*-
 
-import openerp
-from openerp.http import request
-from openerp.osv import osv
-from openerp import SUPERUSER_ID
-from openerp.tools.translate import _
 from datetime import datetime
+import logging
+import simplejson
+import urllib2
+import werkzeug.urls
+
+import openerp
+from openerp.addons.web.http import request
+from openerp import models, api, _
 from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
 
-import werkzeug.urls
-import urllib2
-import simplejson
 
-import logging
 _logger = logging.getLogger(__name__)
 
 
-class google_service(osv.osv_memory):
+class google_service(models.TransientModel):
     _name = 'google.service'
 
-    def generate_refresh_token(self, cr, uid, service, authorization_code, context=None):
-        ir_config = self.pool['ir.config_parameter']
-        client_id = ir_config.get_param(cr, SUPERUSER_ID, 'google_%s_client_id' % service)
-        client_secret = ir_config.get_param(cr, SUPERUSER_ID, 'google_%s_client_secret' % service)
-        redirect_uri = ir_config.get_param(cr, SUPERUSER_ID, 'google_redirect_uri')
+    @api.model
+    def generate_refresh_token(self, service, authorization_code):
+        ir_config = self.env['ir.config_parameter']
+        client_id = ir_config.sudo().get_param('google_%s_client_id' % service)
+        client_secret = ir_config.sudo().get_param('google_%s_client_secret' % service)
+        redirect_uri = ir_config.sudo().get_param('google_redirect_uri')
```

Comment:
Would be simpler to define `ir_config` as `self.env['ir.config_parameter'].sudo()`